pyedflib/edfwriter.py

```diff
@@ -171,39 +176,38 @@ def sex2int(sex):
         raise ValueError(f"Unknown sex: '{sex}'")
 
 
-def gender2int(gender):
+def gender2int(gender: Union[int, str, None]) -> Optional[int]:
     warnings.warn("Function 'gender2int' is deprecated, use 'sex2int' instead.", DeprecationWarning, stacklevel=2)
     return sex2int(gender)
 
 
 class ChannelDoesNotExist(Exception):
-    def __init__(self, value):
+    def __init__(self, value: Any) -> None:
         self.parameter = value
 
-    def __str__(self):
+    def __str__(self) -> str:
         return repr(self.parameter)
 
 
 class WrongInputSize(Exception):
-    def __init__(self, value):
+    def __init__(self, value: Any) -> None:
         self.parameter = value
 
-    def __str__(self):
+    def __str__(self) -> str:
         return repr(self.parameter)
 
 
 class EdfWriter:
-    def __exit__(self, exc_type, exc_val, ex_tb):
+    def __exit__(self, exc_type: Optional[type], exc_val: Optional[BaseException], ex_tb: Optional[BaseException]) -> None:
```

Comment:
```suggestion
    def __exit__(self, exc_type: Optional[type], exc_val: Optional[BaseException], ex_tb: Optional[BaseException]) -> None:
    def __exit__(self, exc_type: Optional[Type[BaseException]], exc_val: Optional[BaseException], exc_tb: Optional[TracebackType]) -> None:
)
    
```

according to https://stackoverflow.com/questions/58808055/what-are-the-python-builtin-exit-argument-types

would also need to import `from types import TracebackType`